Panic when adding annotations to func.yaml

I added annotations in func.yaml and received a panic on deploy. 
All worked find prior to adding the annotations.

func.yaml:
```
specVersion: 0.35.0
name: hello2
runtime: go
registry:
image: docker.io/davidhadas/hello2:latest
imageDigest: sha256:264331febd1dee6360c638d4cd035047a377ff9fcb28b0a7906faebd81992fb2
created: 2023-02-20T15:34:19.042958+02:00
build:
  buildpacks:
  - paketo-buildpacks/go-dist
  - ghcr.io/boson-project/go-function-buildpack:tip
  builder: pack
  buildEnvs: []
run:
  volumes: []
  envs: []
deploy:
  namespace: default
  remote: false
  annotations:
    features.knative.dev/queueproxy-podinfo: enabled
    qpoption.knative.dev/guard-activate: enable
  options: {}
  labels: []
  healthEndpoints:
    liveness: /health/liveness
    readiness: /health/readiness
```

```
kn func deploy 
   🙌 Function image built: docker.io/davidhadas/hello2:latest
🕒 ⬆️  Deploying function to the cluster
E0220 16:26:00.184474    4668 runtime.go:79] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x2f1dd60?, 0x3698ce0})
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0014fda18?})
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x2f1dd60, 0x3698ce0})
	/root/.gvm/gos/go1.19.5/src/runtime/panic.go:884 +0x212
knative.dev/func/knative.updateService.func1(0xc0007f7900)
	/home/prow/go/src/knative.dev/func/knative/deployer.go:405 +0x2c5
knative.dev/client/pkg/serving/v1.updateServiceWithRetry.func1()
	/home/prow/go/src/knative.dev/func/vendor/knative.dev/client/pkg/serving/v1/client.go:279 +0xbe
k8s.io/client-go/util/retry.OnError.func1()
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/client-go/util/retry/util.go:51 +0x33
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x2fb9660, 0x2a32301})
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:222 +0x1b
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x36c22b0?, 0xc000128008?}, 0x50?)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:235 +0x57
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x3094d20?)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:228 +0x39
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x1312d00, 0x3ff8000000000000, 0x3fb999999999999a, 0x3, 0x0}, 0x100d927?)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:423 +0x5f
k8s.io/client-go/util/retry.OnError({0x1312d00, 0x3ff8000000000000, 0x3fb999999999999a, 0x3, 0x0}, 0x33ea560, 0xc0013d8230)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/client-go/util/retry/util.go:50 +0xf1
k8s.io/client-go/util/retry.RetryOnConflict(...)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/client-go/util/retry/util.go:104
knative.dev/client/pkg/serving/v1.updateServiceWithRetry({0x36c2278?, 0xc000fa5200?}, {0x36d8da0?, 0xc000eb3620?}, {0xc0014fda00?, 0x6?}, 0xc00091d280?, 0x3?)
	/home/prow/go/src/knative.dev/func/vendor/knative.dev/client/pkg/serving/v1/client.go:271 +0x1f5
knative.dev/client/pkg/serving/v1.(*knServingClient).UpdateServiceWithRetry(0xc0014fd9e8?, {0x36c2278?, 0xc000fa5200?}, {0xc0014fda00?, 0xc0014fda00?}, 0x6?, 0xc0014fda18?)
	/home/prow/go/src/knative.dev/func/vendor/knative.dev/client/pkg/serving/v1/client.go:262 +0x45
knative.dev/func/knative.(*Deployer).Deploy(_, {_, _}, {{0xc0014fd9e8, 0x6}, {0xc00005c054, 0x4a}, {0xc0014fda00, 0x6}, {0xc0014fda18, ...}, ...})
	/home/prow/go/src/knative.dev/func/knative/deployer.go:258 +0xdf1
knative.dev/func.(*Client).Deploy(0xc000b64d00, {0x36c2278?, 0xc000fa5200}, {0xc00005c054, 0x4a})
	/home/prow/go/src/knative.dev/func/client.go:689 +0x268
knative.dev/func/cmd.runDeploy(0xc00125f800, {0x0?, 0x0?, 0x0?}, 0x33eaba8)
	/home/prow/go/src/knative.dev/func/cmd/deploy.go:327 +0x1665
knative.dev/func/cmd.NewDeployCmd.func1(0xc00125f800?, {0x4bc2a98?, 0x0?, 0x0?})
	/home/prow/go/src/knative.dev/func/cmd/deploy.go:163 +0x25
github.com/spf13/cobra.(*Command).execute(0xc00125f800, {0x4bc2a98, 0x0, 0x0})
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000dec000)
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:961
main.main()
	/home/prow/go/src/knative.dev/func/cmd/func/main.go:41 +0x235
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0014fda18?})
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x2f1dd60, 0x3698ce0})
	/root/.gvm/gos/go1.19.5/src/runtime/panic.go:884 +0x212
knative.dev/func/knative.updateService.func1(0xc0007f7900)
	/home/prow/go/src/knative.dev/func/knative/deployer.go:405 +0x2c5
knative.dev/client/pkg/serving/v1.updateServiceWithRetry.func1()
	/home/prow/go/src/knative.dev/func/vendor/knative.dev/client/pkg/serving/v1/client.go:279 +0xbe
k8s.io/client-go/util/retry.OnError.func1()
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/client-go/util/retry/util.go:51 +0x33
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x2fb9660, 0x2a32301})
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:222 +0x1b
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x36c22b0?, 0xc000128008?}, 0x50?)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:235 +0x57
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x3094d20?)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:228 +0x39
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x1312d00, 0x3ff8000000000000, 0x3fb999999999999a, 0x3, 0x0}, 0x100d927?)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:423 +0x5f
k8s.io/client-go/util/retry.OnError({0x1312d00, 0x3ff8000000000000, 0x3fb999999999999a, 0x3, 0x0}, 0x33ea560, 0xc0013d8230)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/client-go/util/retry/util.go:50 +0xf1
k8s.io/client-go/util/retry.RetryOnConflict(...)
	/home/prow/go/src/knative.dev/func/vendor/k8s.io/client-go/util/retry/util.go:104
knative.dev/client/pkg/serving/v1.updateServiceWithRetry({0x36c2278?, 0xc000fa5200?}, {0x36d8da0?, 0xc000eb3620?}, {0xc0014fda00?, 0x6?}, 0xc00091d280?, 0x3?)
	/home/prow/go/src/knative.dev/func/vendor/knative.dev/client/pkg/serving/v1/client.go:271 +0x1f5
knative.dev/client/pkg/serving/v1.(*knServingClient).UpdateServiceWithRetry(0xc0014fd9e8?, {0x36c2278?, 0xc000fa5200?}, {0xc0014fda00?, 0xc0014fda00?}, 0x6?, 0xc0014fda18?)
	/home/prow/go/src/knative.dev/func/vendor/knative.dev/client/pkg/serving/v1/client.go:262 +0x45
knative.dev/func/knative.(*Deployer).Deploy(_, {_, _}, {{0xc0014fd9e8, 0x6}, {0xc00005c054, 0x4a}, {0xc0014fda00, 0x6}, {0xc0014fda18, ...}, ...})
	/home/prow/go/src/knative.dev/func/knative/deployer.go:258 +0xdf1
knative.dev/func.(*Client).Deploy(0xc000b64d00, {0x36c2278?, 0xc000fa5200}, {0xc00005c054, 0x4a})
	/home/prow/go/src/knative.dev/func/client.go:689 +0x268
knative.dev/func/cmd.runDeploy(0xc00125f800, {0x0?, 0x0?, 0x0?}, 0x33eaba8)
	/home/prow/go/src/knative.dev/func/cmd/deploy.go:327 +0x1665
knative.dev/func/cmd.NewDeployCmd.func1(0xc00125f800?, {0x4bc2a98?, 0x0?, 0x0?})
	/home/prow/go/src/knative.dev/func/cmd/deploy.go:163 +0x25
github.com/spf13/cobra.(*Command).execute(0xc00125f800, {0x4bc2a98, 0x0, 0x0})
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000dec000)
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/home/prow/go/src/knative.dev/func/vendor/github.com/spf13/cobra/command.go:961
main.main()
	/home/prow/go/src/knative.dev/func/cmd/func/main.go:41 +0x235
Error: exit status 2

```

--- a/pkg/knative/deployer.go
+++ b/pkg/knative/deployer.go
@@ -429,6 +429,10 @@ func updateService(f fn.Function, newEnv []corev1.EnvVar, newEnvFrom []corev1.En
 		// this prevents conflicts in Revision name when updating the KService from multiple places.
 		service.Spec.Template.Name = ""
 
+		if service.Spec.Template.ObjectMeta.Annotations == nil {
+			service.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+		}
+
 		// Don't bother being as clever as we are with env variables
 		// Just set the annotations and labels to be whatever we find in func.yaml
 		if decorator != nil {

